src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -50,6 +54,15 @@ export const getAugmentVisSavedObjs = async (
   visId: string | undefined,
   loader: SavedAugmentVisLoader | undefined
 ): Promise<ISavedAugmentVis[]> => {
+  if (visId === undefined || loader === undefined) {
+    return [] as ISavedAugmentVis[];
+  }
+  const config = getUISettings();
+  const isAugmentationEnabled = config.get('visualization:enablePluginAugmentation');
+  if (!isAugmentationEnabled) {
+    // eslint-disable-next-line no-throw-literal
+    throw 'Visualization augmentation is disabled, please enable visualization:enablePluginAugmentation.';
+  }
```

Comment:
I think we can ignore this use case and let this fn assume it is enabled. How about making this check at the top-level in `visualizeEmbeddable` before `fetchVisLayers` is called? This way, everything from `fetchVisLayers` down (including this fn), can ignore having to handle 2 separate scenarios.

Reply:
This function would be called by plugins, so I want to keep this here. In the `fetchVisLayers` function, I check if the setting is enabled and return empty list if it is not enabled.

Reply:
Got it. My original thoughts was this wasn't being used by plugins, but rather as a helper function for vis embeddable to fetch these. Given these other use cases, it makes sense to have these safeguards here.